catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1124,6 +1133,41 @@ private FilterCapabilities buildFilterCapabilities() {
     return filterCapabilities;
   }
 
+  private Set<String> getQueryFilterTransformerTypeNames() {
+    return Stream.of(getBundle())
+        .map(Bundle::getBundleContext)
+        .map(this::getQueryFilterTransformerServices)
+        .flatMap(Collection::stream)
+        .map(
+            queryFilterTransformerServiceReference ->
+                queryFilterTransformerServiceReference.getProperty(
+                    QUERY_FILTER_TRANSFORMER_TYPE_NAMES_FIELD))
+        .filter(Objects::nonNull)
+        .filter(List.class::isInstance)
+        .map(List.class::cast)
+        .flatMap((Function<List, Stream<Object>>) Collection::stream)
+        .filter(String.class::isInstance)
+        .map(String.class::cast)
+        .collect(Collectors.toSet());
+  }
+
+  private Collection<ServiceReference<QueryFilterTransformer>> getQueryFilterTransformerServices(
+      BundleContext bundleContext) {
+    try {
+      return bundleContext.getServiceReferences(QueryFilterTransformer.class, null);
+    } catch (InvalidSyntaxException e) {
+      LOGGER.debug("Unable to get service references for QueryFilterTransformers.", e);
+    }
+    return Collections.emptyList();
+  }
+
+  private List<String> getTypeNames() {
+    return new LinkedList<>(
+        Stream.of(getQueryFilterTransformerTypeNames(), TYPE_NAMES_LIST)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toSet()));
```

Comment:
❗️  Inefficient to create the set just to make distinct and pass to the list. Instead...

```java
    return Stream.of(getQueryFilterTransformerTypeNames(), TYPE_NAMES_LIST)
        .flatMap(Collection::stream)
        .distinct()
        .collect(Collectors.toCollection(LinkedList::new));
```

Reply:
That was my original solution, but others wanted it changed to a set.

Reply:
It's a waste of object creation just to immediate throw it out. I don't understand the change.

Reply:
I believe the other reviewers thought that it was better to use a set to perform the deduplication instead of calling distinct().

Reply:
While the internal implementation of `Stream`'s `distinct` operation is going to use a `HashSet` to perform the dedupe anyway, it's not radically less efficient (in this particular case) but it is butt ugly, to the point where the even the IDE recommends the streaming dedupe followed by the direct collection to `LinkedList`. I'm not going to hold up this PR for this, but I'm quite unhappy with this implementation.

Reply:
@coyotesqrl  Because you have the strongest opinion on this, I'm going to change it back to using `distinct()`. Interestingly, my IntelliJ doesn't flag this, and I generally pay close attention to those IDE warnings.

Reply:
I'm on the latest 2017.3 public preview, ultimate edition of course. Might be a newly added analysis.